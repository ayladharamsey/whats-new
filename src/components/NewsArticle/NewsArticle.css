article {
    background-color: #E4E8C6;
    width: 30%;
    height: 350px;
    padding: 20px;
    display: inline-block;
    margin: 20px;
}

img {
    width: 200px;
    height: 150px;
}