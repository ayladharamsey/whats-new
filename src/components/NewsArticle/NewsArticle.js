import React from 'react';
import './NewsArticle.css';

const NewsArticle = ({ story }) => {
   return (
       <article>
           <img src={story.img} alt={story.headline}></img>
           <h4>{story.headline}</h4>
           <p>{story.description}</p>
           <button> Read More </button>
       </article>
   )
}

export default NewsArticle;