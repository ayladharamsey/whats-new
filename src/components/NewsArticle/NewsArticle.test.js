import React from 'react';
import { shallow } from 'enzyme';
import NewsArticle from './NewsArticle';

describe('NewsArticle', () => {
    let wrapper, storyObject;

    it('should match the snapshot with all data passed correctly', () => {
        storyObject = {
            img : "https://www.cool.com",
            headline :"bad bitches take over the world",
            description :"things are going well",
            id : 2,
            url: "www.cool.com",
            linkOut : jest.fn()
        };
        wrapper = shallow(<NewsArticle story = {storyObject} />) 

        expect(wrapper).toMatchSnapshot();
    });

    it.skip('should call the linkOut function when button is clicked', () => {
        const linkOutMock = jest.fn();
        storyObject = {
            img : "https://www.cool.com",
            headline :"potatos take over the world",
            description :"things are going well",
            id : 2,
            url: "www.cool.com",
            linkOut : {linkOutMock}
        };
        wrapper = shallow(<NewsArticle story = {storyObject} />) 
        wrapper.find('button').simulate('click');
        expect(linkOutMock).toHaveBeenCalledWith(2);
    });
});