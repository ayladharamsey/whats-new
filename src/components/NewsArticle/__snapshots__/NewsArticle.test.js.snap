// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should match the snapshot with all data passed correctly 1`] = `
<article>
  <img
    alt="bad bitches take over the world"
    src="https://www.cool.com"
  />
  <h4>
    bad bitches take over the world
  </h4>
  <p>
    things are going well
  </p>
  <button
    onClick={[Function]}
  >
     Read More 
  </button>
</article>
`;
